client/state/signup/steps/survey/test/reducer.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SIGNUP_STEPS_SURVEY_SET,
+	SIGNUP_COMPLETE_RESET,
+} from 'state/action-types';
+
+import signupSurveyReducer from '../reducer';
+
+describe( 'reducer', () => {
+	it( 'should update the survey', () => {
+		expect( signupSurveyReducer( {}, {
+			type: SIGNUP_STEPS_SURVEY_SET,
+			survey: {
+				vertical: 'test-survey',
+				otherText: 'test-other-text'
+			}
+		} ) ).to.be.eql( {
+			vertical: 'test-survey',
+			otherText: 'test-other-text'
+		} );
+	} );
+
+	it( 'should reset the survey on signup complete', () => {
+		expect( signupSurveyReducer( {}, {
```

Comment:
Shouldn't you pass here as the first arg:

```js
{
  vertical: 'test-survey',
  otherText: 'test-other-text'
}
```

To actually make sure the action is causing the reducer to erase the data?